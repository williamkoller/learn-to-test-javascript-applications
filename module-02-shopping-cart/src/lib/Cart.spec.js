import Cart from './Cart';

describe('Cart', () => {
  it('should return 0 when getTotal() is executed in a newly created instance', () => {
    const cart = new Cart();

    expect(cart.getTotal()).toEqual(0);
  });
});
