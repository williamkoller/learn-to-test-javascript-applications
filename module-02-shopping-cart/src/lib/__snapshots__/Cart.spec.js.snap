// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart checkout() should return and object with total and the list of items 1`] = `
{
  "items": [
    {
      "product": {
        "price": 35388,
        "title": "Adidas running shoed - men",
      },
      "quantity": 2,
    },
    {
      "product": {
        "price": 41872,
        "title": "Adidas running shoed - women",
      },
      "quantity": 2,
    },
  ],
  "total": 154520,
}
`;

exports[`Cart checkout() should return and object with total and the list of items 2`] = `
{
  "items": [
    {
      "product": {
        "price": 35388,
        "title": "Adidas running shoed - men",
      },
      "quantity": 2,
    },
    {
      "product": {
        "price": 41872,
        "title": "Adidas running shoed - women",
      },
      "quantity": 2,
    },
  ],
  "total": 154520,
}
`;

exports[`Cart checkout() should return and object with total and the list of items when summary() 1`] = `
{
  "items": [
    {
      "product": {
        "price": 35388,
        "title": "Adidas running shoed - men",
      },
      "quantity": 2,
    },
    {
      "product": {
        "price": 41872,
        "title": "Adidas running shoed - women",
      },
      "quantity": 2,
    },
  ],
  "total": 154520,
}
`;

exports[`Cart checkout() should return and object with total and the list of items when summary() is called 1`] = `
{
  "items": [
    {
      "product": {
        "price": 35388,
        "title": "Adidas running shoed - men",
      },
      "quantity": 2,
    },
    {
      "product": {
        "price": 41872,
        "title": "Adidas running shoed - women",
      },
      "quantity": 2,
    },
  ],
  "total": 154520,
}
`;
